cmd/create/cluster/shared_vpc.go

```diff
@@ -0,0 +1,96 @@
+package cluster
+
+import (
+	"fmt"
+
+	"github.com/openshift/rosa/pkg/aws"
+	"github.com/openshift/rosa/pkg/interactive"
+	"github.com/openshift/rosa/pkg/rosa"
+	"github.com/spf13/cobra"
+	errors "github.com/zgalor/weberr"
+)
+
+func isSubnetBelongToSharedVpc(r *rosa.Runtime, accountID string, subnetIDs []string,
+	mapSubnetToOwnerID map[string]string) bool {
+	for _, subnetID := range subnetIDs {
+		ownerID := mapSubnetToOwnerID[subnetID]
+		if ownerID != "" && ownerID != accountID {
+			r.Reporter.Infof(fmt.Sprintf("Subnet with ID '%s' is shared by AWS account '%s', "+
+				"Defaulting to installing a cluster into a shared VPC.", subnetID, ownerID))
+			return true
+		}
+	}
+
+	return false
+}
+
+func getPrivateHostedZoneID(cmd *cobra.Command, privateHostedZoneID string) (string, error) {
+	res, err := interactive.GetString(interactive.Input{
+		Question: "Private hosted zone ID",
+		Help:     cmd.Flags().Lookup("private-hosted-zone-id").Usage,
+		Default:  privateHostedZoneID,
+		Required: true,
+	})
+	if err != nil {
+		return "", errors.Errorf("Expected a valid value for 'private-hosted-zone-id': %s", err)
+	}
+
+	return res, nil
+}
+
+func getSharedVpcRoleArn(cmd *cobra.Command, sharedVpcRoleArn string) (string, error) {
+	res, err := interactive.GetString(interactive.Input{
+		Question: "Shared VPC role ARN",
+		Help:     cmd.Flags().Lookup("shared-vpc-role-arn").Usage,
+		Default:  sharedVpcRoleArn,
+		Required: true,
+		Validators: []interactive.Validator{
+			aws.ARNValidator,
+		},
+	})
+	if err != nil {
+		return "", errors.Errorf("Expected a valid value for 'shared-vpc-role-arn': %s", err)
+	}
+
+	return res, nil
+}
+
+func getBaseDomain(r *rosa.Runtime, cmd *cobra.Command, baseDomain string) (string, error) {
+	dnsDomains, err := getAvailableBaseDomains(r)
+	if err != nil {
+		return "", err
+	}
+
+	res, err := interactive.GetOption(interactive.Input{
+		Question: "Base Domain",
+		Help:     cmd.Flags().Lookup("base-domain").Usage,
+		Default:  baseDomain,
```

Comment:
should we validate the provided baseDomain by users exist or not, like the current logic for subnet ids?

Reply:
If the user is in an interactive mode, they can choose only from the list of DNS domains retrieved from the backend.
So if they provide a domain that doesn't exist, they won't be able to continue. 
Do you mean checking for the option of providing a base domain using a flag?

Reply:
I mean the similar code in subnet ids here: https://github.com/openshift/rosa/blob/ca5ffb4daa2b3748975f8eea1c2d6a9a14855a6a/cmd/create/cluster/cmd.go#L1859-L1873

Reply:
Since we have validation in the backend, I think it's ok.
Just thinking about duplicate login across the CLI and the backend.
Also, the error message returned from the backend will include more details in case the base domain exists but is associated with another cluster.